src/TenorGuitar/Page.purs

```diff
@@ -0,0 +1,573 @@
+        let
+          fstring = fingeredString { x, y }
+        _ <- H.modify
+          ( \st -> st
+              { mouseDownFinger = Just fstring
+              , errorText = ""
+              , pitches = []
+              }
+          )
+        pure unit
+      else do
+        _ <- H.modify
+          ( \st -> st
+              { mouseDownFinger = Nothing
+              , pitches = []
+              }
+          )
+        pure unit
+    MouseUp cx cy -> do
+      state <- H.get
+      let
+        x = toNumber cx - state.canvasPosition.left
+        y = toNumber cy - state.canvasPosition.top
+      if (y > titleDepth) then do
+        let
+          fstring = fingeredString { x, y }
+        _ <- H.modify (\st -> st { mouseUpFinger = Just fstring, errorText = "" })
+        _ <- handleQuery (EditFingering unit)
+        pure unit
+      else do
+        _ <- H.modify (\st -> st { mouseUpFinger = Nothing })
+        pure unit
+    GetChordName name -> do
+      state <- H.get
+      let
+        newShape = state.chordShape { name = name }
+        newState = state
+          { chordShape = newShape
+          , errorText = ""
+          , pitches = []
+          }
+      _ <- H.put newState
+      _ <- handleQuery (DisplayFingering unit)
+      pure unit
+    GetFirstFretNumber numStr -> do
+      state <- H.get
+      let
+        fret = fromMaybe 0 $ fromString numStr
+        newShape = state.chordShape { firstFretOffset = fret }
+        newState = state { chordShape = newShape, errorText = "" }
+      _ <- H.put newState
+      _ <- handleQuery (DisplayFingering unit)
+      pure unit
+    ClearFingering -> do
+      _ <- H.modify
+        ( \st -> st
+            { chordShape = openStringChordShape
+            , errorText = ""
+            , pitches = []
+            }
+        )
+      _ <- handleQuery (DisplayFingering unit)
+      pure unit
+    GetImageScale scale -> do
+      _ <- H.modify (\st -> st { exportScale = scale })
+      pure unit
+    Export format -> do
+      state <- H.get
+      let
+        originalCanvas = unsafePartial (fromJust state.mCanvas)
+        mimeType = toMimeType format
+        scaleFactor = toNumber state.exportScale / 100.0
+        fileName = (safeName state.chordShape.name) <> "_tenor_guitar"
+      canvas <- H.liftEffect $ scaleCanvas originalCanvas scaleFactor
+      _ <- H.liftEffect $ exportAs canvas fileName mimeType
+      pure unit
+    Load (FIC.FileLoaded filespec) -> do
+      state <- H.get
+      let
+        validated = validateJson filespec.contents
+        newState = validation
+          (\errs -> state { errorText = foldl (<>) "" errs })
+          (\chordShape -> state { chordShape = chordShape, errorText = "" })
+          validated
+      _ <- H.put newState
+      _ <- handleQuery (DisplayFingering unit)
+      pure unit
+    Save -> do
+      state <- H.get
+      let
+        name = (safeName state.chordShape.name) <> "_guitar" <> ".json"
```

Comment:
Sorry - missed this one - we need tenor guitar rather than guitar here too.

Reply:
Updated. Thanks for catching that one.